test/unit/workDiv/src/FooVec.hpp

```diff
@@ -0,0 +1,152 @@
+/* Copyright 2022 Sergei Bastrakov, Jan Stephan, Mehmet Yusufoglu
+ * SPDX-License-Identifier: MPL-2.0
+ */
+
+#pragma once
+
+#include <alpaka/dim/Traits.hpp>
+#include <alpaka/elem/Traits.hpp>
+#include <alpaka/extent/Traits.hpp>
+#include <alpaka/vec/Vec.hpp>
+
+#include <array>
+#include <iostream>
+
+//! User defined vector for testing the usability of any vector type.
+//!
+//! \tparam TVec The data type.
+//! \tparam Size Vector size as a non-type parameter.
+template<typename TVec, std::size_t Size>
+class FooVec
+{
+public:
+    static_assert(Size <= 3, "Size must be 3 or smaller");
+    std::array<TVec, Size> arr;
+
+    // Default Constructor
+    FooVec()
+    {
+        arr.fill(TVec());
+    }
+
+    // Constructor with initializer list
+    FooVec(std::initializer_list<TVec> initList)
+    {
+        if(initList.size() <= Size)
+        {
+            std::copy(initList.begin(), initList.end(), arr.begin());
+        }
+        else
+        {
+            throw std::out_of_range("Initializer list size exceeds array size");
+        }
+    }
+
+    // Destructor
+    ~FooVec()
+    {
+        // No explicit cleanup required for this example
+    }
+
+    // Copy Constructor
+    FooVec(FooVec const& other)
+    {
+        arr = other.arr;
+    }
+
+    // Copy Assignment Operator
+    FooVec& operator=(FooVec const& other)
+    {
+        if(this != &other)
+        {
+            arr = other.arr;
+        }
+        return *this;
+    }
+
+    // Move Assignment Operator
+    FooVec& operator=(FooVec&& other) noexcept
+    {
+        if(this != &other)
+        {
+            arr = std::move(other.arr);
+        }
+        return *this;
+    }
```

Comment:
```suggestion
```
I don't see a reason for implementing special member functions.

Reply:
Ok removed. Might be misleading. Thanks.